@(notation: String, proteinName: String, accession: String, biol: ArrayList[Biolsource], protein: List[Proteins], uniprot: List[String], gssites: List[GsProteinStr2], source: HashSet[Reference], description: List[SitesReferences], sequence: String, proteinMultiple: List[Proteins], generalSites: List[GeneralSites], definedSites: List[DefinedSites], typeEntry: String, swissProtName: String, proteinsource: List[Proteinsource])


@main {


  <ul class="breadcrumb">
    <li><i class="icon-home" ></i><a href="#"> UniCarbKB</a> <span class="divider">></span></li>
    <li><i class="icon-map-marker" ></i><a href="/browse"> Protein</a> <span class="divider">></span></li>
    <li class="active" > @proteinName</li>
  </ul>

  <section id="layouts" class="proteinsummary">

    <div class="page-header row-fluid">
      <h1>Associated Structures</h1>
      <h4 class="subheader span8">Accession: 
	@if(typeEntry.matches("not swiss prot")) {<a href="http://uniprot.org/uniprot/?query=@accession+AND+reviewed%3Ayes&sort=score">@accession </a><span class="pull-right"><a href="http://uniprot.org/uniprot/?query=@accession+AND+reviewed%3Ayes&sort=score"><span class='label label-light'>UniProtKB/Swiss-Prot</span></a></span> }
     	@if(!typeEntry.matches("not swiss prot")) {<a href="http://uniprot.org/uniprot/@swissProtName">@accession </a><span class="pull-right"><a href="http://uniprot.org/uniprot/@accession"><span class='label label-light'>UniProt Entry</span></a></span> }
      </h4>
    </div>


    <div class="row-fluid">
      <div class="span8">
	<div class='name'>
          @* <div class="control-group">
            <label>Name:</label> <div class="controls">@proteinName</div>
          </div>
	  *@
        </div> 



        <div class='ptm-details'>
	  <h3>UniProtKB/Swiss-Prot PTM Description</h3>
	    @for(uniprot <- uniprot) {
		@if(uniprot.matches("No info")) {  <div class="alert alert-error">No UniProt PTM information available for this protein entry </div> }
		@if(!uniprot.matches("No info")) {    
              	<p>@uniprot</p>}
	    }
        </div>

        <div class='glycosylation-sites'>
	  @if(generalSites.size() > 0 ) {
	  <h3>Glycosylation Sites</h3>
	  @for(generalSites <- generalSites) {
	  @if(generalSites.strSiteGeneral.size() > 0 ) {
	  <table class="computers table table-striped">
            <thead>
              <tr>
                <th>Position</th>
                <th>Structures</th>
                <th>Description</th>
                <th>Evidence</th>
              </tr>
            </thead>

            <tbody>
	    <tr>
                <td>@generalSites.glyco_aa_site</td>
		<td>
		  <a href="/proteinsite?position=@generalSites.glyco_aa_site&protein=@accession&type=general">
		  <span class="label label-notice"><i class='icon-th icon-white'></i> 
                      Associated Structures: <b>@generalSites.strSiteGeneral.size()</b></span>
		  </a>
		</td>
		<td><span class="label">global</span></td>
                <td>GlycoSuite</td>
	    </tr>
	    </tbody>
   	  </table>
	  }}}
        </div>

	<div class='glycosylation-sites'>
          @if(definedSites.size() > 0 ) {
	  <h5>Site Specific Information</h5>
	  <p>A number of glycan structures have been assigned to specific glycosylation sites</p>
          <table class="computers table table-striped">
            <thead>
              <tr>
                <th>Position</th>
                <th>Structures</th>
                <th>Description</th>
                <th>Evidence</th>
              </tr>
            </thead>

            <tbody>
            @for(definedSites <- definedSites) {
            <tr>
                <td>@definedSites.amino_acid_position</td>
                <td>
		  <a href="/proteinsite?position=@definedSites.amino_acid_position&protein=@accession&type=defined">
                  <span class="label label-notice"><i class='icon-th icon-white'></i>
                      Associated Structures: <b>@definedSites.strSiteDefined.size()</b></span>
		  </a>
                </td>
                <td><span class="label">Site specific</span></td>
                <td>GlycoSuite</td>
            </tr>
            }
            </tbody>
          </table>
	  }
        </div>


	<div class='glycosylation-sites'>
          <h3>Glycan Structures</h3>
          <ul class="structures clearfix">
          
      	  @for(s <- proteinMultiple) {
          @for((s, i) <- s.stproteins.zipWithIndex; if (i <=5) ){
          <li>
	  @views.html.format.structure(notation, s.structure.id)
          </li>
          }}
          </ul>
          <ul id='more-thumbnails' class="structures clearfix">
          @for(s <- proteinMultiple) { 
          @for(s <- s.stproteins ){
          <li class="span5">
          <a href="@routes.Application.structureDetails(s.structure.id)">
            <img src="../images/@notation/_@{s.structure.id}.png" alt="">
          </a>
          </li>
          }}
          </ul>
         <div id='show-structures' class='more-structures' href='#'>
       		See more structures
          <br />
          <span>&#9679; &#9679; &#9679;</span>
         </div>
        </div>



        <div class='notes'>
          <h3>Notes</h3>
          <p>Accompanying GlycoSuite information</p>
          @if(!description.isEmpty() ) {
          @for(d <- description){
          <ol>
            <li>@d.glyco_aa_site</li>
          </ol>
          }}
        </div>

        <div class='history'>
          <h3>History</h3>
          <label>Integrated:</label>
          <div class="controls">2010-03-01</div>
          <label>Last updated:</label>
          <div class="controls">2012-11-02</div>
        </div>

      </div><!-- /col -->

      <div class="span4 details">
	
	@views.html.format.format()

	@if(sequence.length > 1) {
        <div class="info">
          <h3>Sequence</h3>
	  		<div class="sequence">
            @sequence
          	</div>
        </div>}
        
        <div class="info">
      		<div class='info'>
        	<h3>Biological Associations</h4>
        	<div class='source'>
          	<a id='toggle-source'><span class='label label-success'><span class='icon-leaf icon-white'></span> Taxonomy (@proteinsource.size()) <span class="caret"></span></span></a>
            </div>
        <div>
          <ul id='more-source'>
            <h3 id='less-source'><span class='icon-tags icon-white'></span> Taxonomies</h3>
            @for(p <- proteinsource){
            <li><span class='icon-tag icon-white'></span> @p.species @p.system </li>
            }
          </ul>   
        </div>
      	</div>
     

        <div class="info">
          <h3>References @if(source.size() > 5) {showing top 5} <span class="label label-notice" style="font-size:16px">@source.size()</span> </h3>
          <ol>
	    @for((source,i) <- source.zipWithIndex; if (i < 5)) {
	    <li class="references">
		<p class="title"><a href='/references/@source.id'>@source.title</a></p>
                <p class="author">@source.authors</p>
                <p class="ref">PubMed: <a href='http://www.ncbi.nlm.nih.gov/pubmed/@source.pmid'>@source.pmid</a></p>
            </li>
	    }
          </ol>
          @*
          @for(b <- biol) {
          @b.sourceref

          }
          *@
          @*
          <ol>
            <li class="references">
              <p class="title">"Molecular cloning, sequence, and specificity of expression of the gene encoding the low molecular weight human salivary mucin (MUC7)."</p>
              <p class="author"><a>Bobek L.A.</a>, <a>Tsai H.</a>, <a>Biesbrock A.R.</a>, <a>Levine M.J.</a></p>
              <p class="ref">J. Biol. Chem. 268:20563-20569(1993) [PubMed: 7690757]</p>
            </li>

            <li class="references">
              <p class="title">"Low-molecular-mass human salivary mucin, MG2: structure and binding of Pseudomonas aeruginosa."</p>
              <p class="author"><a>Reddy M.S.</a>, <a>Levine M.J.</a>, <a>Paranchych W.</a></p>
              <p class="ref">Crit. Rev. Oral Biol. Med. 4:315-323(1993) [PubMed: 8104046]</p>
            </li>

            <li class="references">
              <p class="title">"The status, quality, and expansion of the NIH full-length cDNA project: the Mammalian Gene Collection (MGC)."</p>
              <p class="author"><a>The MGC Project Team</a></p>
              <p class="ref">Genome Res. 14:2121-2127(2004) [PubMed: 15489334]</p>
            </li>

            <li class="references">
              <p class="title">"Generation and annotation of the DNA sequences of human chromosomes 2 and 4."</p>
              <p class="author"><a>Hillier L.W.</a>, <a>Graves T.A.</a>, <a>Fulton R.S.</a>, <a>Fulton L.A.</a>, <a>Pepin K.H.</a>, <a>Minx P.</a>, <a>Wagner-McPherson C.</a>, <a>Layman D.</a>, <a>Wylie K.</a>, <a>Sekhon M.</a>, <a>Becker M.C.</a>, <a>Fewell G.A.</a>, <a>Delehaunty K.D.</a>, <a>Miner T.L.</a>, <a>Nash W.E.</a>, <a>Kremitzki C.</a>, <a>Oddy L.</a>, <a>Du H.  Wilson R.K.</a></p>
              <p class="ref">Nature 434:724-731(2005) [PubMed: 15815621]</p>
            </li>

            <li class="references">
              <p class="title">"Molecular cloning, sequence, and specificity of expression of the gene encoding the low molecular weight human salivary mucin (MUC7)."</p>
              <p class="author"><a>Bobek L.A.</a>, <a>Tsai H.</a>, <a>Biesbrock A.R.</a>, <a>Levine M.J.</a></p>
              <p class="ref">J. Biol. Chem. 268:20563-20569(1993) [PubMed: 7690757</p>
            </li>
          </ol>
          *@
        </div>

      </div><!-- /col -->
    </div><!-- /row -->

    @views.html.footerunicarb.footerunicarb()

  </section>

  }
